### DISCUSSION TITLE
MRTK Scene system not optimal for VR development

### DISCUSSION THREAD

**Discussion by jelmerS2** [microsoft/MixedRealityToolkit-Unity#9509] [Q&A] [answered]
We're using MRTK2.6 in a Oculus Quest project, together with the MRTK scene system. I thought I just leave my 2 cents here, just some thoughts on the scene system and Vr development. 

It seems that the MRTK Scene System is designed with HoloLens in mind, which has some implications for VR. 

The main problem is lighting and active scene behaviour: by default MRTK proposes to use a separate scene with some default lights which becomes the "active" active scene (there can only be one active scene at the time in Unity, this scene is responsible for not only lighting but also the background and other properties such as fog I believe). 
This is a very strange workflow for general game development, because normally you include lighting and background in your game scene, not separately. 

One can disable this behaviour in MRTK so that there is no separate "lighting scene", but then the ManagerScene (or the scene containing the MRTK object) remains the active scene, thus still overriding the game scenes lighting and other things. This is a not optimal for VR dev.

Our workaround has been very simple though. We include this snippet of code in every "game scene". So far there seem to be no side effects but what worries me is that the active scene is also used for new GameObjects instantiated by scripts, [according to the unity docs](https://docs.unity3d.com/ScriptReference/SceneManagement.SceneManager.SetActiveScene.html). So this could potentially backfire for us later =) 

```
using UnityEngine;
using UnityEngine.SceneManagement;

namespace Assets.Scripts.Core
{
	public class SetActiveScene : MonoBehaviour
	{
		void Start()
		{	
			SceneManager.SetActiveScene(gameObject.scene);
#if DEBUG
			Debug.Log($"Set {gameObject.scene.name} as active scene");
#endif
		}
	}
}
```

But I think it would be good to reconsider how the MRTK scene manager works to make MRTK a bit more logical for VR projects.

**Comment by jelmerS2** [comment]
code tags don't work?

**Reply by timGerken** [reply]
Triple back-tick for code tags :)
> \`\`\` C#
> { code }
> \`\`\`
``` C#
using UnityEngine;
using UnityEngine.SceneManagement;

namespace Assets.Scripts.Core
{
  public class SetActiveScene : MonoBehaviour
  {
    void Start()
    {
      SceneManager.SetActiveScene(gameObject.scene);
#if DEBUG
      Debug.Log($"Set {gameObject.scene.name} as active scene");
#endif
    }
  }
}
```

**Comment by david-c-kline** [comment]
@Railboy. Since you originated the scene system, do you have any thoughts on this?

**Comment by david-c-kline** [comment, marked as answer]
@jelmerS2,

We have converted this discussion into an issue (#9600) so that we can track it in our v3.x work. Please feel free to provide suggestions, insights, opinions regarding whether or not the MRTK scene system should be removed in v3, etc.

Thanks!
